### PR TITLE
feat(performance): Add key transactions as a column in performance

### PR DESCRIPTION
**Pull request by Zylphrex**
This change removes by key transactions pill on the performance page and
replaces it with a key transactions column. Since key transactions is a boolean
column, we also enforce a secondary sort of transaction name then project name
when sorting by key transactions.

# Screenshot
image

### CODE REVIEW

**Review by markstory — COMMENTED** [review]
From the screenshot you posted, you might want to use emotion to make the key transaction column less wide. It seems larger than it needs to be right now.

File: src/sentry/static/sentry/app/views/performance/landing.tsx [hunk 1]
```diff
@@ -265,44 +261,29 @@ class PerformanceLanding extends React.Component<Props, State> {
   }
 
   renderHeaderButtons() {
+    const {organization} = this.props;
+    const views: FilterViews[] = [FilterViews.ALL_TRANSACTIONS];
+    if (!organization.features.includes('key-transactions')) {
```
Comment (dashed) [inline src/sentry/static/sentry/app/views/performance/landing.tsx:266]:
```suggestion
    if (organization.features.includes('key-transactions')) {
```

I think this is a typo.

Comment (dashed) [reply src/sentry/static/sentry/app/views/performance/landing.tsx:266]: Ah. Nevermind.

File: src/sentry/static/sentry/app/views/performance/table.tsx [hunk 2]
```diff
@@ -211,23 +251,50 @@ class Table extends React.Component<Props, State> {
...
+
+    // We special case sort by key transactions here to include
+    // the transaction name and project as the secondary sorts.
+    const keyTransactionSort = eventView.sorts.find(
+      sort => sort.field === 'key_transaction'
+    );
+    if (keyTransactionSort) {
+      const allowedFields = eventView.getFields();
+      const sorts = ['key_transaction', 'transaction', 'project']
+        .filter(field => allowedFields.includes(field))
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/performance/table.tsx:265]: Do you need this step? Isn't the field list for performance table fixed?

**Review by markstory — APPROVED** [review]

**Review by dashed — APPROVED** [review]

File: src/sentry/static/sentry/app/views/performance/table.tsx (hunk 2 above)
Comment (Zylphrex) [reply src/sentry/static/sentry/app/views/performance/table.tsx:265]: It was just to be on the safe side, but I can remove it.

**Pull request merged**